src/engine/controls/cuecontrol.cpp

```diff
@@ -567,6 +657,45 @@ void CueControl::hotcueSet(HotcueControl* pControl, double v) {
     }
 }
 
+void CueControl::hotcueSetLoop(HotcueControl* pControl, double v) {
+    //qDebug() << "CueControl::hotcueSetLoop" << v;
+
+    if (!v)
+        return;
+
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack)
+        return;
+
+    int hotcue = pControl->getHotcueNumber();
+    // Note: the cue is just detached from the hotcue control
+    // It remains in the database for later use
+    // TODO: find a rule, that allows us to delete the cue as well
+    // https://bugs.launchpad.net/mixxx/+bug/1653276
+    hotcueClear(pControl, v);
+
+    CuePointer pCue(m_pLoadedTrack->createAndAddCue());
+    double loopStartPosition = m_pLoopStartPosition->get();
+    double loopLength = m_pLoopEndPosition->get() - loopStartPosition;
+    if (loopStartPosition < 0 || loopLength <= 0) {
+        return;
+    }
+
+    pCue->setPosition(loopStartPosition);
+    pCue->setLength(loopLength);
+    pCue->setHotCue(hotcue);
+    pCue->setLabel("");
+    pCue->setType(Cue::CUE);
```

Comment:
Cue::LOOP

Reply:
Please see: https://github.com/mixxxdj/mixxx/pull/2194#issuecomment-520772333

Reply:
I understand how this can simplify the code in some ways. However, in the long run the current approach will be both confusing and limiting. Currently the code distinguishes between normal hotcues and loop cues by whether they have a nonzero length. This prohibits us from ever having nonzero length hotcues in the future, which could be useful for user-defined section markers. Also, the current approach requires two different cue type enums, which is confusing. 

Using the single existing type enum which is already stored in the database will require implementing an abstraction layer to collect multiple types into a single ordered list. This would require storing that ordered list in a new column in the cues database table because we could no longer infer the hotcue number from the order of the cues in the database or along the timeline of the track. This would also lay the foundation for making more types of cues accessible via the hotcue_X_* Controls if we want to do that in the future.

Reply:
Oh, we already store the hotcue number in the database.